src/elisp/treemacs-macros.el

```diff
@@ -134,7 +134,7 @@ Will bind to current button to 'btn' for the executon of the action forms.
 When NO-ERROR is non-nil no error will be thrown if no match for the button
 state is achieved."
   (declare (debug (&rest [sexp form])))
-  `(-if-let (btn (treemacs-current-button))
+  `(-if-let (btn treemacs--last-selected-button)
```

Comment:
This change breaks treemacs, ensuring that TAB only applies to the last clicked button. This whole thing seems to be more complicated than it first appeared. I think it's best if I take over from here.

Reply:
Sure, you know the code better than me. You can take over.

Reply:
Pushed my changes. Please make sure it works on your end as well, and I'll merge.

Reply:
Just tested it. Works.